File: app/src/main/java/com/jameschamberlain/footballteamtracker/fixtures/EditRecyclerAdapter.java
package com.jameschamberlain.footballteamtracker.fixtures;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.jameschamberlain.footballteamtracker.R;

import java.util.ArrayList;

public class EditRecyclerAdapter extends RecyclerView.Adapter<EditRecyclerAdapter.ViewHolder> {

    private Fixture fixture;
    private ArrayList<String> players;
    private boolean isGoals;

    EditRecyclerAdapter(Fixture fixture, boolean isGoals) {
        this.fixture = fixture;
        if (isGoals) {
            this.players = fixture.getGoalscorers();
        }
        else {
            this.players = fixture.getAssists();
        }
        this.isGoals = isGoals;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_player_edit, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        String player = players.get(position);
        holder.name.setText(player);
        holder.imageView.setOnClickListener(new EditOnClickListener(fixture, isGoals, player, this));
    }

    @Override
    public int getItemCount() {
        return players.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        ImageView imageView;
        TextView name;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            imageView = itemView.findViewById(R.id.clear_image_view);
            name = itemView.findViewById(R.id.name_text_view);
        }
    }
}

File: app/src/main/java/com/jameschamberlain/footballteamtracker/fixtures/EditRecyclerAdapter.kt
package com.jameschamberlain.footballteamtracker.fixtures

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.jameschamberlain.footballteamtracker.databinding.ItemPlayerEditBinding
import java.util.*

class EditRecyclerAdapter internal constructor(private val fixture: Fixture, private val isGoals: Boolean) : RecyclerView.Adapter<EditRecyclerAdapter.ViewHolder>() {
    private var players: ArrayList<String> = if (isGoals) {
        fixture.goalscorers
    } else {
        fixture.assists
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemBinding = ItemPlayerEditBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(itemBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val player = players[position]
        holder.nameTextView.text = player
        holder.clearImageView.setOnClickListener(EditOnClickListener(fixture, isGoals, player, this))
    }

    override fun getItemCount(): Int {
        return players.size
    }

    inner class ViewHolder(itemBinding: ItemPlayerEditBinding) : RecyclerView.ViewHolder(itemBinding.root) {
        var nameTextView: TextView = itemBinding.nameTextView
        var clearImageView: ImageView = itemBinding.clearImageView
    }

}